ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/Authorizer.scala

```diff
@@ -48,9 +71,15 @@ final class Authorizer(now: () => Instant, ledgerId: String, participantId: Stri
       applicationId: Option[String],
       call: (Req, StreamObserver[Res]) => Unit): (Req, StreamObserver[Res]) => Unit =
     authorize(call) { claims =>
-      valid(claims) &&
-      parties.forall(claims.canReadAs) &&
-      applicationId.forall(claims.validForApplication)
+      for {
+        _ <- valid(claims)
+        _ <- parties.foldLeft[Either[AuthorizationError, Unit]](Right(()))((acc, e) =>
+          acc.flatMap(_ => claims.canReadAs(e)))
+        _ <- applicationId.fold[Either[AuthorizationError, Unit]](Right(()))(id =>
+          claims.validForApplication(id))
```

Comment:
not using fold but `.map(...).getOrElse(...)` let's you get rid of the explicit type annotation, because `validForApplication(id)` already determines the type:
```scala
_ <- applicationId.map(claims.validForApplication).getOrElse(Right()))
``` 

might still be a long line though.